### DISCUSSION TITLE
Other slots are ignored in full page components

### DISCUSSION THREAD

**Discussion by mokhosh** [livewire/livewire#3012] [7. Bugs (archived)]
### Description
If you have a full page component that uses `layouts.app` for example and you have `slot` and `header` let's say, any `wire:` directive in the `header` slot will be ignored.

### Exact steps to reproduce
1. Fresh Laravel install + Livewire
2. Create a full page component that uses default layout
3. Create some method
4. Put a button inside `slot="header"` and put `wire:click` on it and try to call your method

**Comment by ryangjchandler** [comment]
@mokhosh This is somewhat expected behaviour. With a full-page component, that `header` slot is going to be outside of the main Livewire component's root (inside of the default `slot`).

**Comment by mokhosh** [comment]
I thought so, but I also thought we might think of a way to track what's defined inside the template, not what's inside the tag.

**Comment by joshhanley** [comment]
@mokhosh I solve it by making sure everything is inside a layout component, rather than in the layout file, and wrapping your component views in that layout component. That means then everything, including the page header can be controlled by Livewire.

Have a look here for more details https://joshhanley.com.au/articles/how-to-structure-your-layout-file-for-livewire

Hope it helps!
